### PR TITLE
Adds the stake details card

### PR DESCRIPTION
**Pull request by georgeweiler**

### CODE REVIEW

File: src/pages/Staking/StakeDetailsPage/index.tsx [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+  const { stakingProviderAddress } = useParams()
+  const stake = useSelector((state: RootState) =>
+    selectStakeByStakingProvider(state, stakingProviderAddress!)
+  ) as StakeData
+
+  const isInActiveStake = stake
+    ? BigNumber.from(stake?.totalInTStake).isZero()
+    : false
+
+  const rewardsForStake = useFetchRewardsForSingleStake(stake?.stakingProvider)
```
Comment (michalsmiarowski) [inline src/pages/Staking/StakeDetailsPage/index.tsx:37]: To be honest I don't understand why we create hook for fetching rewards for single stake 🤔 We have `useFetchStakingRewards` hook that fetches rewards for all stakes and is executed in the `App.tsx`. Do we really need that hook? Can't we just pull the data from the store?

**Review by michalsmiarowski — CHANGES_REQUESTED** [review]
Left some comments to look at before the merge 👀

File: src/pages/Staking/StakeDetailsPage/StakeDetailRow.tsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+}> = ({ label, value, isPrimary }) => {
+  return (
+    <HStack justify="space-between" minH="40px">
+      <BoxLabel status={isPrimary ? "primary" : "secondary"}>{label}</BoxLabel>
+      {typeof value === "string" ? (
+        <CopyToClipboard textToCopy={value}>
+          <BodyMd color={isPrimary ? "brand.500" : "gray.500"}>
+            {shortenAddress(value)}
+          </BodyMd>
+        </CopyToClipboard>
```
[the remark this answers is not in the document]
Comment (georgeweiler) [reply src/pages/Staking/StakeDetailsPage/StakeDetailRow.tsx:19]: Let's fix after launch. There are a lot of small tech debt and design debt items like that that I would like to tackle.

File: src/pages/Staking/index.tsx [hunk 3]
```diff
@@ -70,6 +66,15 @@ const StakingProviderDetails: PageComponent = (props) => {
...
 
+  const selectedTabId = useMemo(() => {
+    if (pathname.includes("details")) {
+      return "1"
+    }
+    if (pathname.includes("authorize")) {
+      return "2"
+    }
+  }, [pathname])
+
```
Comment (r-czajkowski) [inline src/pages/Staking/index.tsx:77]: We need to figure out a better solution for the next nested level of navigation in page components. The same pattern we are going to use in the `how it works` page and probably in other places in the future. Ofc non-blocking comments, just pointing this out.

**Review by r-czajkowski — CHANGES_REQUESTED** [review]

File: src/pages/Staking/StakeDetailsPage/NodeStatusLabel.tsx [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+          icon: FiAlertCircle,
+          color: "red.500",
+        }
+    }
+  }, [status])
+
+  return (
+    <BoxLabel status={status} icon={<Icon as={icon} />}>
+      {text}
+    </BoxLabel>
```
[the remark this answers is not in the document]
Comment (georgeweiler) [reply src/pages/Staking/StakeDetailsPage/NodeStatusLabel.tsx]: 👍  updated.

File: src/pages/Staking/index.tsx (hunk 3 above)
Comment (georgeweiler) [reply src/pages/Staking/index.tsx:77]: I agree. The filter tabs need a way to recognize when they are link components so they can automatically handle this. You're right - I've implemented this same pattern in the how-it-works page here: https://github.com/threshold-network/token-dashboard/pull/198/files#diff-2662b5a0473a36cacbb7559a3fa2ac27daef984cdb79caf6fb2d9825add8d479R34-R45 


I am not a fan of this pattern but I think we should fix the filter tab after launch.

File: src/pages/Staking/StakeDetailsPage/StakeDetailRow.tsx [hunk 5]
```diff
@@ -0,0 +1,27 @@
+import { FC } from "react"
+import { BodyMd, BoxLabel, HStack } from "@threshold-network/components"
+import CopyToClipboard from "../../../components/CopyToClipboard"
+import shortenAddress from "../../../utils/shortenAddress"
+
+const StakeDetailRow: FC<{
+  label: string
+  value: string | JSX.Element
+  isPrimary?: boolean
```
[the remark this answers is not in the document]
Comment (georgeweiler) [reply src/pages/Staking/StakeDetailsPage/StakeDetailRow.tsx]: 1b2d18542868b038ccc933e1f14470d22e9d3b5e

**Review by r-czajkowski — APPROVED** [review]
LGTM! 🚀

**Review by michalsmiarowski — APPROVED** [review]
LGTM 🤸‍♂️

**Pull request merged**